packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -11,3 +37,97 @@ export const makeNobleAccount = (async (orch: Orchestrator) => {
   return nobleChain.makeAccount();
 }) satisfies OrchestrationFlow;
 harden(makeNobleAccount);
+
+export const forwardFunds = async (
+  orch: Orchestrator,
+  {
+    currentChainReference,
+    supportsCctp,
+    log = makeTracer('FlowForwardFunds'),
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+  }: Context,
+  tx: {
+    txHash: EvmHash;
+    amount: NatAmount;
+    destination: AccountId;
+  },
+) => {
+  await null;
+  assertAllDefined({
+    currentChainReference,
+    supportsCctp,
+    log,
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+    tx,
+  });
+  const { amount, destination, txHash } = tx;
+  log('trying forward for', amount, 'to', destination, 'for', txHash);
+
+  const { namespace, reference } = parseAccountId(destination);
+
+  const settlement = await settlementAccount;
+  const intermediateRecipient = getNobleICA().getAddress();
+
+  if (reference === currentChainReference) {
+    await settlement.send(destination, amount);
+    statusManager.forwarded(txHash);
+  } else if (namespace === 'cosmos') {
+    try {
+      await settlement.transfer(destination, amount, {
+        timeoutRelativeSeconds: FORWARD_TIMEOUT.sec,
+        forwardOpts: {
+          intermediateRecipient,
+          timeout: FORWARD_TIMEOUT.p,
+        },
+      });
+      log('forward retry successful for', txHash);
```

Comment:
This can be a first time forward attempt too, no ? Actually right now without retry it can only be a first time. Why "retry" ?

Reply:
From when the scope of the change was the retry feature instead of just a refactor. I'll revise